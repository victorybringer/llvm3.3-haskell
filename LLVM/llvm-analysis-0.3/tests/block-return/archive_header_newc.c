extern int find_newc_header(void);
extern void* __archive_read_ahead(void);
extern void __archive_read_consume(void);
extern int rand(void);

extern int g;

int header_newc(void) {
  int r;
  void *h;

  r = find_newc_header();
  if(r < 100)
    return r;

  h = __archive_read_ahead();
  if(h == 0)
    return -10;

  __archive_read_consume();

  if(rand())
    g = 0;
  else if(rand())
    g = 1001;
  else {}

  __archive_read_consume();

  return r;
}
