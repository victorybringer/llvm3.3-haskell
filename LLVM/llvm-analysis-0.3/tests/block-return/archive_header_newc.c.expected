fromList [ ("%if.then3", -10) ]
